Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.py

```diff
@@ -2049,16 +2161,35 @@ def get_system_status(client, command_args) -> Tuple[str, dict, Union[list, dict
     ipaddrs_server = argToList(arg=command_args.get('ip_server'))
     port = arg_to_number(arg=command_args.get('port'))
 
-    is_resp_filtering_required = statuses or hostnames or ipaddrs_client or ipaddrs_client or ipaddrs_server or port
+    # is_resp_filtering_required = statuses or hostnames or ipaddrs_client or ipaddrs_client or ipaddrs_server or port
+    is_resp_filtering_required = statuses or ipaddrs_client or ipaddrs_client or ipaddrs_server or port
+
     filter_arguments = [
         (statuses, 'status'),
-        (hostnames, 'host_name'),
         (ipaddrs_client, 'ipaddress_client'),
         (ipaddrs_server, 'ipaddress_server'),
         ([port], 'port_number')
     ]
 
-    raw_response = client.do_request('GET', '/api/v2/system_status')
+    cache_filters = []
+    if hostnames:
+        for hostname in hostnames:
+            cache_filters.append({
+                "field": "host_name",
+                "value": f"^{hostname}[A-Za-z.]*$",
+                "operator": "RegexMatch"
```

Comment:
Do you have a link to the documentation that you used?

Reply:
The documentation was provided from Tanium via their API documentation. I can email you the PDF to this documentation, but you can find the specific docs I used listed under "Objects - cache_filters". There I took note of the different cache filters we can apply to the API query, one of which is a regex match. The regex I used will match a string to the name provided by the user as long as it appears in the beginning of the string (for example, I can provide jboyce and it will match my hostname of jboyce.example.net)